executor/merge_join.go

```diff
@@ -323,12 +302,20 @@ func (e *MergeJoinExec) fetchNextInnerRows() (err error) {
 	return nil
 }
 
-func (e *MergeJoinExec) fetchNextOuterRows() (err error) {
-	e.outerRows, err = e.outerIter.rowsWithSameKey()
+// fetchNextOuterRows fetches the next Chunk of outer table. Rows in a Chunk
```

Comment:
fetchNextOuterRows fetches the Next Chunk from outer table.
All the rows in this Chunk are not promised to have the same join key,
but promised to be sorted according to the join keys.